packages/cells/src/cells/date-picker-cell.tsx

```diff
@@ -1,68 +1,126 @@
-            }
-        }
-        return (
-            <input
-                style={{ minHeight: 26, border: "none", outline: "none" }}
-                type={format}
-                autoFocus={true}
-                value={val}
-                onChange={e => {
-                    p.onChange({
-                        ...p.value,
-                        data: {
-                            ...p.value.data,
-                            date: e.target.valueAsDate ?? undefined,
-                        },
-                    });
-                }}
-            />
-        );
-    },
-    onPaste: (v, d) => {
-        let newDate: Date | undefined;
-        try {
-            newDate = new Date(v);
-        } catch {
-            /* do nothing */
+export type DatePickerCell = CustomCell<DatePickerCellProps>;
+
+const Editor: ReturnType<ProvideEditorCallback<DatePickerCell>> = (cell) => {
+  console.log(cell);
+  const cellData = cell.value.data;
+  const { min, max, step, readonly, format, displayDate } = cellData;
+  const value = formatValueForHTMLInput(format, cellData.date);
+  if (readonly) {
+    return (
+      <TextCellEntry
+        highlight={true}
+        autoFocus={false}
+        disabled={true}
+        value={displayDate ?? ""}
+        onChange={() => undefined}
+      />
+    );
+  }
+  return (
+    <input
+      data-testid={"test-id"}
+      required
+      style={{ minHeight: 26, border: "none", outline: "none" }}
+      type={format}
+      value={value}
+      min={min}
+      max={max}
+      step={step}
+      autoFocus={true}
+      onChange={(event) => {
+        if (event.target.value === "") {
+          cell.onChange({
+            ...cell.value,
+            data: {
+              ...cell.value.data,
+              // just set the value to undefined if submitted (enter or clicking out)
+              // escape still works
+              date: undefined,
+            },
+          });
+        } else {
+          cell.onChange({
+            ...cell.value,
+            data: {
+              ...cell.value.data,
+              // use valueAsNumber because valueAsDate is null for "datetime-local"
+              // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#technical_summary
+              date: new Date(event.target.valueAsNumber) ?? cellData.date,
+            },
+          });
         }
+      }}
+    />
+  );
+};
 
-        return {
-            ...d,
-            date: Number.isNaN(newDate) ? undefined : newDate,
-        };
-    },
+const renderer: CustomRenderer<DatePickerCell> = {
+  kind: GridCellKind.Custom,
+  isMatch: (cell: CustomCell): cell is DatePickerCell =>
+    (cell.data as any).kind === "date-picker-cell",
+  draw: (args, cell) => {
+    const { displayDate } = cell.data;
+    drawTextCell(args, displayDate, cell.contentAlign);
+    return true;
+  },
+  provideEditor: () => ({
+    editor: Editor,
+  }),
+  onPaste: (v, d) => {
```

Comment:
Maybe this should support time-only formats as well in `onPaste`. Without using another library, the best option might be to try converting the pasted value into a full date string (in case `newDate` is Nan/undefined):

```
new Date(`1970-01-01T${v}Z`)
```